apiserver/src/main/java/org/dependencytrack/resources/v1/AdvisoriesResource.java

```diff
@@ -0,0 +1,380 @@
+                        getAlpineRequest(),
+                        handle -> handle.attach(AdvisoryDao.class)
+                                .getAdvisoriesWithFindingsByProject(project.getId(),
+                                        suppressed));
+                final long totalCount = advisoryWithFindingRows.size();
+
+                return Response.ok(advisoryWithFindingRows.stream().toList())
+                        .header(TOTAL_COUNT_HEADER, totalCount).build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND)
+                        .entity("The project could not be found.").build();
+            }
+        }
+    }
+
+    @GET
+    @Path("/project/{projectId}/advisory/{advisoryId}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Returns a list of findings associated to project x advisory", description = "<p>Requires permission <strong>VIEW_VULNERABILITY</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "A list of all advisory findings for a specific project and advisory", content = {
+                    @Content(array = @ArraySchema(schema = @Schema(implementation = AdvisoryDao.ProjectAdvisoryFindingRow.class)), mediaType = MediaType.APPLICATION_JSON),
+            }),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "403", description = "Access to the requested project is forbidden", content = @Content(schema = @Schema(implementation = ProblemDetails.class), mediaType = ProblemDetails.MEDIA_TYPE_JSON)),
+            @ApiResponse(responseCode = "404", description = "The project or advisory could not be found")
+    })
+    @PaginatedApi
+    @PermissionRequired(Permissions.Constants.VIEW_VULNERABILITY)
+    public Response getFindingsByProjectAdvisory(
+            @Parameter(description = "The ID of the project", schema = @Schema(type = "string"), required = true) @PathParam("projectId") long projectId,
+            @Parameter(description = "The advisoryId", schema = @Schema(type = "string"), required = true) @PathParam("advisoryId") long advisoryId,
+            @HeaderParam("accept") String acceptHeader) {
+        try (QueryManager qm = new QueryManager(getAlpineRequest())) {
+            LOGGER.info("Querying for " + projectId + " :: " + advisoryId);
+            List<AdvisoryDao.ProjectAdvisoryFindingRow> advisoryRows = withJdbiHandle(getAlpineRequest(),
+                    handle -> handle.attach(AdvisoryDao.class)
+                            .getFindingsByProjectAdvisory(projectId, advisoryId));
+            final long totalCount = advisoryRows.size();
+            LOGGER.info("retrieved size " + totalCount);
+
+            return Response.ok(advisoryRows.stream().toList()).header(TOTAL_COUNT_HEADER, totalCount)
+                    .build();
+        }
```

Comment:
Use either the JDO API (`QueryManager`), *or* JDBI, but don't mix them in the same block. They cannot share transactions or connections. Keeping a handle of both of them open wastes resources.